packages/frontend/src/components/Routing.js

```diff
@@ -564,23 +588,10 @@ class Routing extends Component {
                                     );
                                 }}
                             />
-                            <Route exact path="/batch-import" render={(({location}) => {
-                                let { keys, accounts, ledgerHdPaths } = parse(location.hash, {arrayFormat: 'comma'});
-                                if (!keys || !accounts) {
-                                    return <PageNotFound />;
-                                }
-
-                                // if single key or account param make an array of it
-                                keys = Array.isArray(keys) ? keys : [keys];
-                                accounts = Array.isArray(accounts) ? accounts : [accounts];
-                                ledgerHdPaths = Array.isArray(ledgerHdPaths) ? ledgerHdPaths : [ledgerHdPaths];
-
-                                const accountIdToKeyMap = accounts.reduce((acc, curr) => {
-                                    const [ accountId, keyIndex, ledgerHdPathIndex ] = curr.split('*');
-                                    return { ...acc, [accountId]: {key: keys[keyIndex], ledgerHdPath: ledgerHdPaths?.[ledgerHdPathIndex]} };
-                                }, {});
-                                return <BatchImportAccounts accountIdToKeyMap={accountIdToKeyMap} onCancel={() => this.props.history.replace('/')}/>;
-                            })} />
+                            <Route exact path="/batch-import" render={() =>
+                                (<BatchImportAccounts
+                                    onCancel={() => this.props.history.replace('/')} />)}
```

Comment:
Not sure I see why `onCancel` needs to be passed here rather than be a function within the component?

Reply:
I think no doesn't matter where declarate `history.replace`, here, on top-level component, or inside FC.